Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -427,33 +427,29 @@ namespace gdjs {
 
     _checkGrabPlatform() {
       const object = this.owner;
+      let oldX = object.getX();
+      let oldY = object.getY();
+
       let tryGrabbingPlatform = false;
-      object.setX(
-        object.getX() +
-          (this._requestedDeltaX > 0
-            ? this._xGrabTolerance
-            : -this._xGrabTolerance)
-      );
-      let collidingPlatform = this._getCollidingPlatform();
-      if (collidingPlatform !== null && this._canGrab(collidingPlatform)) {
-        tryGrabbingPlatform = true;
-      }
-      object.setX(
-        object.getX() +
-          (this._requestedDeltaX > 0
-            ? -this._xGrabTolerance
-            : this._xGrabTolerance)
-      );
+      for (let i = 0; i < this._potentialCollidingObjects.length; ++i) {
+        const platform = this._potentialCollidingObjects[i];
 
-      //Check if we can grab the collided platform
-      if (tryGrabbingPlatform) {
-        let oldY = object.getY();
+        object.setX(
```

Comment:
Do you think we could do this setX only once, get all the colliding platforms (i.e: we can do a `_getCollidingPlatforms` method), then we iterate on all of them to see if there is at least one that we can grab. Then we put back the object to its original X position (oldX, like you did), and we either bail out (if no platform to grab) or grab the first platform that can be grabbed (in the future we could add an option to force a decision, like the highest platform on the Y axis).

In other words, rather than:
```
for each potential colliding platform:
  move the object on X axis
  check the collision
  move back on X axis
  grab the platform if we can (i.e: set the Y position and check no collision with other platforms)
```

we would do:

```
move the object on X axis
for each potential colliding platform:
  check the collision (adding them to a "potential grabbing platform" array)
move back on X axis

for each potential grabbing platform
  grab the platform if we can (i.e: set the Y position and check no collision with other platforms)
```

The reason why I'm asking is because this could avoid "moving" the object too much, which can be costly as this invalidates the hitboxes and need to recompute them for collision tests.

Reply:
Ok, the changes are pushed.